crates/wasmtime/src/runtime/externals/table.rs

```diff
@@ -220,12 +225,12 @@ impl Table {
     /// # Panics
     ///
     /// Panics if `store` does not own this table.
-    pub fn size(&self, store: impl AsContext) -> u32 {
+    pub fn size(&self, store: impl AsContext) -> u64 {
         self.internal_size(store.as_context().0)
     }
 
-    pub(crate) fn internal_size(&self, store: &StoreOpaque) -> u32 {
-        unsafe { (*store[self.0].definition).current_elements }
+    pub(crate) fn internal_size(&self, store: &StoreOpaque) -> u64 {
+        unsafe { (*store[self.0].definition).current_elements as u64 }
```

Comment:
Could this be `u64::try_from(...).unwrap()`? Here the `.unwrap()` should be ok because the runtiem should always guarantee that we can fit the number of elements in a 64-bit integer.